oci/internal/oci_util.go

```diff
@@ -0,0 +1,121 @@
+package internal
+
+import (
+	"github.com/pkg/errors"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+)
+
+// annotation spex from https://github.com/opencontainers/image-spec/blob/master/annotations.md#pre-defined-annotation-keys
+const (
+	separator = `(?:[-._:@+]|--)`
+	alphanum  = `(?:[A-Za-z0-9]+)`
+	component = `(?:` + alphanum + `(?:` + separator + alphanum + `)*)`
+)
+
+var refRegexp = regexp.MustCompile(`^` + component + `(?:/` + component + `)*$`)
+var windowsRefRegexp = regexp.MustCompile(`^([a-zA-Z]:\\.+?):(.*)$`)
+
+// ValidImageName returns true is the specified image matches the open-containers image name specs, false otherwise.
+func ValidImageName(image string) bool {
+	if len(image) == 0 {
+		return true
+	}
+	return refRegexp.MatchString(image)
+}
+
+// SplitDirAndImage tries to split the provided OCI reference into the OCI directory and image.
+// The image is returned as string pointer, allowing for nil being returned, if no image part could be extraced from the reference.
+// Neither directory nor image parts are validated at this stage.
+func SplitDirAndImage(reference string) (string, string) {
```

Comment:
Please rename this `dir` to `path`, throughout.

The comment about a string pointer is no longer true.

Reply:
+1